drivers/media/v4l2-core/v4l2-mem2mem.c

```diff
@@ -301,8 +301,9 @@ static void __v4l2_m2m_try_queue(struct v4l2_m2m_dev *m2m_dev,
 
 	dprintk("Trying to schedule a job for m2m_ctx: %p\n", m2m_ctx);
 
-	if (!m2m_ctx->out_q_ctx.q.streaming
-	    || !m2m_ctx->cap_q_ctx.q.streaming) {
+	if (!m2m_ctx->out_q_ctx.buffered &&
+	    (!m2m_ctx->out_q_ctx.q.streaming ||
+	     !m2m_ctx->cap_q_ctx.q.streaming)) {
```

Comment:
I may have misunderstood the intention of this commit, but it appears to bypass the streaming check if running in buffered mode.

Reply:
You're right, but it wasn't quite my intent. Too much inverted logic.

I wanted the existing logic when not buffering, but to not return should either queue be streaming when buffered. One to rework.